solr/core/src/test/org/apache/solr/handler/component/DistributedCombinedQueryComponentTest.java

```diff
@@ -0,0 +1,301 @@
+                    + "\"params\":{\"combiner\":true,\"combiner.query\":[\"lexical1\"]}}",
+                "shards",
+                getShardsString(),
+                CommonParams.QT,
+                "/search"));
+    assertEquals(1, rsp.getResults().size());
+    assertFieldValues(rsp.getResults(), id, "2");
+  }
+
+  @Override
+  protected String getShardsString() {
+    if (deadServers == null) return shards;
+    Arrays.sort(shardsArr);
+    StringBuilder sb = new StringBuilder();
+    for (String shard : shardsArr) {
+      if (!sb.isEmpty()) sb.append(',');
+      sb.append(shard);
+    }
+    return sb.toString();
+  }
+
+  /**
+   * Tests multiple lexical queries using the Solr server.
+   *
+   * @throws Exception if any error occurs during the test execution
+   */
+  @Test
+  public void testMultipleLexicalQuery() throws Exception {
+    prepareIndexDocs();
+    QueryResponse rsp;
+    rsp =
+        queryServer(
+            createParams(
+                CommonParams.JSON,
+                "{\"queries\":"
+                    + "{\"lexical1\":{\"lucene\":{\"query\":\"title:title test for doc 1\"}},"
+                    + "\"lexical2\":{\"lucene\":{\"query\":\"text:test text for doc 2\"}}},"
+                    + "\"limit\":5,"
+                    + "\"fields\":[\"id\",\"score\",\"title\"],"
+                    + "\"params\":{\"combiner\":true,\"combiner.query\":[\"lexical1\",\"lexical2\"]}}",
+                "shards",
+                getShardsString(),
+                CommonParams.QT,
+                "/search"));
+    assertEquals(5, rsp.getResults().size());
+    assertFieldValues(rsp.getResults(), id, "2", "1", "4", "5", "8");
+  }
+
+  /**
+   * Test multiple query execution with sort.
+   *
+   * @throws Exception the exception
+   */
+  @Test
+  public void testMultipleQueryWithSort() throws Exception {
+    prepareIndexDocs();
+    QueryResponse rsp;
+    rsp =
+        queryServer(
+            createParams(
+                CommonParams.JSON,
+                "{\"queries\":"
+                    + "{\"lexical1\":{\"lucene\":{\"query\":\"title:title test for doc 1\"}},"
+                    + "\"lexical2\":{\"lucene\":{\"query\":\"text:test text for doc 2\"}}},"
+                    + "\"limit\":5,\"sort\":\"mod3_idv desc\""
+                    + "\"fields\":[\"id\",\"score\",\"title\"],"
+                    + "\"params\":{\"combiner\":true,\"combiner.query\":[\"lexical1\",\"lexical2\"]}}",
+                "shards",
+                getShardsString(),
+                CommonParams.QT,
+                "/search"));
+    assertEquals(5, rsp.getResults().size());
+    assertFieldValues(rsp.getResults(), id, "2", "8", "5", "4", "1");
+  }
+
+  /**
+   * Tests the hybrid query functionality of the system.
+   *
+   * @throws Exception if any unexpected error occurs during the test execution.
+   */
+  @Test
+  public void testHybridQueryWithPagination() throws Exception {
```

Comment:
This test (or another test?) should be more comprehensive so that it's easier to validate that pagination is in fact working.  I have doubts based on the code, and the test itself doesn't help at all.
For example, first do the query without pagination (return ~5 docs).  Then (same test!), do again (same query) with pagination using an offset & limit that slice into the first request.  And maybe one more time; different page.  It'll be straight-forward to reason about valid pagination results given the first set of results.
As a reviewer I look at the current test as-is and I have no idea if it's working.
I don't care if the test is "hybrid" or not, so long as there are 2 sub-queries thus exercising typical logic.